plugins/usbdmx/USBDMXCom.cpp

```diff
@@ -0,0 +1,272 @@
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * USBDMXCom.cpp
+ * The synchronous and asynchronous USBDMXCom widgets.
+ * Copyright (C) 2021 Peter Newman
+ */
+
+#include "plugins/usbdmx/USBDMXCom.h"
+
+#include <string.h>
+#include <string>
+
+#include "libs/usb/LibUsbAdaptor.h"
+#include "ola/Constants.h"
+#include "ola/Logging.h"
+#include "ola/StringUtils.h"
+#include "ola/util/Utils.h"
+#include "plugins/usbdmx/AsyncUsbSender.h"
+#include "plugins/usbdmx/ThreadedUsbSender.h"
+
+namespace ola {
+namespace plugin {
+namespace usbdmx {
+
+using std::string;
+using ola::usb::LibUsbAdaptor;
+
+namespace {
+
+// Why is this so long?
+static const unsigned int URB_TIMEOUT_MS = 500;
+static const unsigned char ENDPOINT = 0x02;
+enum { USBDMXCOM_MAX_FRAME_SIZE = (512*3)+5 };
+
+/*
+ * Create a USBDMX.com message to match the supplied DmxBuffer.
+ */
+size_t CreateFrame(const DmxBuffer &buffer, uint8_t frame[USBDMXCOM_MAX_FRAME_SIZE]) {
+  static uint8_t old_Values[512];
+  static bool first_use = true;
+  size_t frame_length = 0;
+
+  if (first_use) {
+    first_use = false;
+    OLA_INFO << "Sending 'Tx ON' message to USBDMX.COM adapter";
+    frame[frame_length++] = 0x26; // No-Op
+    frame[frame_length++] = 0x26; // No-Op
+    frame[frame_length++] = 0x26; // No-Op
+    frame[frame_length++] = 0x26; // No-Op
+    frame[frame_length++] = 0x44; // DMX TX ON
```

Comment:
Can we have all these magic numbers as constants please!

Reply:
> Some initial comments.
> 
> Let me know if you want assistance on the todo items, and particularly for DMX input whether you want to merge in the current state and add DMX input in a separate PR or get it all done in one?



> Some initial comments.
> 
> Let me know if you want assistance on the todo items, and particularly for DMX input 

Oh yes please, assistance is much appreciated, as OLA is complex for me, and also, i'm new to the PR workflow!

> whether you want to merge in the current state and add DMX input in a separate PR or get it all done in one?

Maybe little steps by little steps?

Reply:
> Can we have all these magic numbers as constants please!

At least this one should be an easy fix :)